include/Instrument.h

```
@@ -14,98 +14,138 @@
   int exposure;
   int device;
-  double airmass;      // airmass and apcorr used for nightly priors
+  double airmass; // airmass and apcorr used for nightly priors
   double apcorr;
-  double magshift;	// Additive adjustment to all incoming mags (exposure time)
+  double magshift; // Additive adjustment to all incoming mags (exposure time)
 
-  string wcsName;      // Name of final WCS (and map into field coordinates)
-  string mapName;      // Name of photometry map or astrometric map into exposure coords
-  T1* map;	       // The map from pixel coordinates to field coordinates.
-  astrometry::Wcs* wcs;       // Wcs from pixel coordinates to sky coordinates = basemap + field projection
-  std::unique_ptr<astrometry::Wcs> startWcs;  // Input Wcs for this extension (owned by this class)
-  bool needsColor;	// Save info on whether map requires color information.
+  string wcsName;                            // Name of final WCS (and map into field coordinates)
+  string mapName;                            // Name of photometry map or astrometric map into exposure coords
+  T1 *map;                                   // The map from pixel coordinates to field coordinates.
+  astrometry::Wcs *wcs;                      // Wcs from pixel coordinates to sky coordinates = basemap + field projection
+  std::unique_ptr<astrometry::Wcs> startWcs; // Input Wcs for this extension (owned by this class)
+  bool needsColor;                           // Save info on whether map requires color information.
 
-  std::map<long, T2*> keepers; // The objects from this Extension catalog that we will use
+  std::map<long, T2 *> keepers; // The objects from this Extension catalog that we will use
+  void addWcs(string wcsin)
+  {
+    istringstream iss(wcsin);
+    astrometry::PixelMapCollection pmcTemp;
+    if (!pmcTemp.read(iss))
+    {
+      cerr << "Could not deserialize starting WCS" << endl;
+      exit(1);
```

If this is ever going to be called by Python, we'll want it to throw an exception rather than print and exit.

---

This _was_ added for the python interface, but ExtensionBase is no longer wrapped at all, so I'll delete this function.